model/pdata/common.go

```diff
@@ -908,3 +908,10 @@ func (sm StringMap) Sort() StringMap {
 func newStringKeyValue(k, v string) otlpcommon.StringKeyValue { //nolint:staticcheck // SA1019 ignore this!
 	return otlpcommon.StringKeyValue{Key: k, Value: v} //nolint:staticcheck // SA1019 ignore this!
 }
+
+// Sizer returns the size of a Traces, Metrics, or Logs.
+type Sizer interface {
+	// Size returns the size in bytes of a Traces, Metrics or Logs.
+	// If the type is not known, an error will be returned.
+	Size(v interface{}) (int, error)
+}
```

Comment:
You should have a "TracesSizer" similar with "TracesMarshaler"

Reply:
Ahh that makes a lot more sense. I've created `TracesSizer`, `MetricsSizer`, and `LogsSizer` :)